### DISCUSSION TITLE
DS18B20 Support or workaround

### DISCUSSION THREAD

**Discussion by tawright915** [meshtastic/Discussions#63] [General]
I've seen where there was a decision not to support this sensor anymore, but wondered if there was a workaround if there is an need for it, like in my case.  I need an external waterproof temp sensor, and have only seen the DS18B20 probe sensors.

**Comment by garthvh** [comment]
SHT3x and SHT4x sensors are I2C and supported by meshtastic

**Reply by thebentern** [reply]
Also DS2484 is on my list of hardware to add support for, which translates the 1-wire to I2C for these DS18B20 probes

**Reply by tawright915** [reply]
It seems that most of the SHT3x and 4x sensors are not waterproof, nor a probe.  I'm working on a project that will take and send water temperature over a period of time and I do not think that a temp sensor inside of a waterproof case will give me an accurate reading.

**Reply by caveman99** [reply]
DS2484 is an adapter board that will convert the DS18B20 to be compatible with meshtastic i2c sensor system. See https://www.adafruit.com/product/5976

**Reply by thebentern** [reply]
> It seems that most of the SHT3x and 4x sensors are not waterproof, nor a probe. I'm working on a project that will take and send water temperature over a period of time and I do not think that a temp sensor inside of a waterproof case will give me an accurate reading.

There's some waterproof probes with those modules inside on AliExpress. Haven't seen them much in domestic markets here in the US though.

**Reply by tawright915** [reply]
> DS2484 is an adapter board that will convert the DS18B20 to be compatible with meshtastic i2c sensor system. See https://www.adafruit.com/product/5976

So I received the DS2484 and attached the temp probe and wired it to the HelTec board and got nothing.  I turned on the environmental settings in the configuration.  But no temp is showing up on the main screen.  Am I doing something wrong?  Do I need to do the configuration through the laptop and not the iPad?  
![IMG_5486](https://github.com/user-attachments/assets/bfc4de8e-57f6-48af-8a40-ed4588fe011c)

**Reply by tawright915** [reply]
> Also DS2484 is on my list of hardware to add support for, which translates the 1-wire to I2C for these DS18B20 probes

So does this mean that it is currently not supported?

**Reply by garthvh** [reply]
> > Also DS2484 is on my list of hardware to add support for, which translates the 1-wire to I2C for these DS18B20 probes
> 
> So does this mean that it is currently not supported?

There is no support for analog sensors, supported sensors need to be I2C

**Reply by tawright915** [reply]
> > > Also DS2484 is on my list of hardware to add support for, which translates the 1-wire to I2C for these DS18B20 probes
> > 
> > 
> > So does this mean that it is currently not supported?
> 
> There is no support for analog sensors, supported sensors need to be I2C

So using the DS2484 is analog?  Maybe I'm not understanding.  In this thread another person mentioned that getting the DS2484 board, and connecting the probe to it would allow the probe to work by converting it to I2C.  Which is what I've setup in the image above.

**Reply by garthvh** [reply]
It is not supported yet

**Reply by tawright915** [reply]
> It is not supported yet

I guess that statement by caveman99 is a little misleading in that I assumed that while it may convert the signal, it still is not compatible.  I went ahead and ordered this sensor - https://a.co/d/bHZZMSz. Do you think this will work better?

**Reply by tawright915** [reply]
@garthvh - Thanks for all the help.  I was able to find a temp probe that works and is waterproof.  I'm not sure if I shared what this project is about, but I coach a FIRST Lego League robotics team.  This is a team of kids from 5th to 7th grade.  Each year they have to define a problem and come up with a solution based on FLL's theme.  This year the theme is problems people in marine type careers might experience.  The kids felt that marine ecologists and biologists have trouble monitoring the levels of phytoplankton in the ocean, which is the basis of the food chain and provides 50% of the oxygen we breath.  They found that certain environment variables over the years has caused the phytoplankton to decrease by 40%.  After doign some research about different trackers that scientists use for tracking sharks, and other species, they looked to ESP32  LORA and Meshtastic.  Their first version was a PVC tube and, foam insulation that you put around pipes in the winter time, to hold the ESP board and battery, and mount the temp probe out the bottom of the tube with the antenna out the top, and using a bunch of these launch them into the ocean to track temp changes.  They plan on using a PH sensor and light sensor to, down the road.  So this help in getting their first iteration off to a good start.  
**Any feedback from this forum would be great.**  

Regardless of if this will work or not, they just need to impress the panel of judges they need to stand before to present their project.  And for a bunch of 5th to 7th graders that is the most intimidating part of the whole thing.  

They plan on sharing any of the feedback with the judges.
